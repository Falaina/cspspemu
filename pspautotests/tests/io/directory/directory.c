#include <common.h>

#include <pspsdk.h>
#include <pspkernel.h>
#include <pspthreadman.h>
#include <psploadexec.h>

int main(int argc, char **argv) {
	SceIoDirent file;
	int fd;
	int result;

	fd = sceIoDopen("folder");
	while (1) {
		if ((result = sceIoDread(fd, &file)) <= 0) break;
		printf(
			"%d:'%s':%lld:%d:%d\n",
			result,
			file.d_name,
			file.d_stat.st_size,
			file.d_stat.st_attr,
			file.d_stat.st_mode
		);
	}
	sceIoDclose(fd);

	return 0;
}