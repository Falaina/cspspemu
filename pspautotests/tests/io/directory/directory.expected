7:'.':0:16:4607
6:'..':0:16:4607
5:'1.TXT':5:32:8703
4:'2.TXT':6:32:8703
3:'TEST.TXT':15:32:8703
2:'1':0:16:4607
1:'subfolder':0:16:4607
